nix/devtools.nix

```diff
@@ -14,7 +14,18 @@ let
       ''
         rootdir="$(${git}/bin/git rev-parse --show-toplevel)"
 
-        ${silver-searcher}/bin/ag -l . "$rootdir" | ${entr}/bin/entr -r "$@"
+        watch() {
+          # invert error code: if entr returns with an error it's because a new file has been added or deleted
+          # if entr returns with exit code 0 it's because the user canceled via CTRL+C
+          trap 'return 0' ERR
+          ${silver-searcher}/bin/ag -l . "$rootdir" | ${entr}/bin/entr -dr "$@"
+          trap - ERR
+          return 1
+        }
+
+        while true; do
+          watch "$@"
```

Comment:
@monacoremo Can you have a look at this?

This allows detection of added and deleted files via `postgrest-watch`.

Reply:
Looks good if it works! Not able to test it right now... I'm a bit puzzeled why entr is behaving like this, the plain loop worked well for me previously

Reply:
> the plain loop worked well for me previously

Thanks for the hint. This made me have another look. My function is essentially doing nothing else than negating the exit code of entr. TIL about the bash `!` operator - which does exactly the same. So instead of the whole function I should be able to just write:
```
while true; do
  ! ${silver-searcher}/bin/....
done
```

So you were only missing one character in your first approach :).

Note: I just tested it and needed 3 characters, because the whole command has to be wrapped in `(! ...)` to work.